ci/.gitlab-ci.yml

```diff
@@ -19,7 +19,7 @@ stages:
     - trying
   variables:
     GIT_SUBMODULE_STRATEGY: recursive
-    SPACK_SHA: 08efe13efb627793fbe048542890ee85668c8100
+    SPACK_SHA: e4f4b70ed153167c06aa63eefa41d975394af066
```

Comment:
Should we add a comment here that specifies why we are using a custom version of spack?

Reply:
FIY `pika@0.9.0` is available in official spack, so the only missing thing is the fix for `umpire`, specifically `CMAKE_HIP_ARCHITECTURES`, for which there is an open PR by @msimberg https://github.com/spack/spack/pull/32025

Reply:
I would just change the comment, I don't want to keep this on hold 😄 thanks!

Reply:
Indeed, I wouldn't wait for that to be merged. I've sent some messages to try to get it in a bit faster but don't hold your breath.